tfx/examples/cifar10/cifar10_utils_native_keras.py

```diff
@@ -0,0 +1,338 @@
+              is_train: bool,
+              batch_size: int = 200) -> tf.data.Dataset:
+  """Generates features and label for tuning/training.
+
+  Args:
+    file_pattern: List of paths or patterns of input tfrecord files.
+    tf_transform_output: A TFTransformOutput.
+    batch_size: representing the number of consecutive elements of returned
+      dataset to combine in a single batch
+
+  Returns:
+    A dataset that contains (features, indices) tuple where features is a
+      dictionary of Tensors, and indices is a single Tensor of label indices.
+  """
+  transformed_feature_spec = (
+      tf_transform_output.transformed_feature_spec().copy())
+  dataset = tf.data.experimental.make_batched_features_dataset(
+      file_pattern=file_pattern,
+      batch_size=batch_size,
+      features=transformed_feature_spec,
+      reader=_gzip_reader_fn,
+      label_key=transformed_name(LABEL_KEY))
+
+  # Apply data augmentation
+  if is_train:
+    dataset = dataset.map(lambda x, y: (_data_augmentation(x), y))
+
+  return dataset
+
+def _freeze_model_by_percentage(model: tf.keras.Model,
+                                percentage: float):
+  """Freeze part of the model based on specified percentage
+
+  Args:
+    model: The keras model need to be partially frozen
+    percentage: the percentage of layers to freeze
+  """
+  if percentage < 0 or percentage > 1:
+    raise Exception("freeze percentage should between 0.0 and 1.0")
+
+  num_layers = len(model.layers)
+  num_layers_to_freeze = int(num_layers * percentage)
+  for idx, layer in enumerate(model.layers):
+    if idx < num_layers_to_freeze:
+      layer.trainable = False
+    else:
+      layer.trainable = True
+
+def _build_keras_model() -> tf.keras.Model:
+  """Creates a Image classification model with MobileNet as backbone
+
+  Returns:
+    The image classifcation Keras Model and the backbone MobileNet model
+  """
+  # We create a MobileNet model with weights pre-trained on ImageNet.
+  # We remove the top classification layer of the MobileNet, which was
+  # used for classifying ImageNet objects. We will add our own classification
+  # layer for CIFAR10 later. We use average pooling at the last convolution
+  # layer to get a 1D vector for classifcation, which is consistent with the
+  # origin MobileNet setup
+  base_model = tf.keras.applications.MobileNet(
+      input_shape=(224, 224, 3), include_top=False, weights='imagenet',
+      pooling='avg')
+
+  # We add a Dropout layer at the top of MobileNet backbone we just created to prevent
+  # overfiting, and then a Dense layer to classifying CIFAR10 objects
+  model = tf.keras.Sequential([
+      tf.keras.layers.InputLayer(
+          input_shape=(224, 224, 3), name=transformed_name(IMAGE_KEY)),
+      base_model,
+      tf.keras.layers.Dropout(0.3),
+      tf.keras.layers.Dense(10, activation='softmax')
+  ])
+
+  return model, base_model
+
+def _data_augmentation(feature_dict):
```

Comment:
We can tighten this a bit and just take an input tensor that has a batch of images.